apps/prairielearn/python/prairielearn/core.py

```diff
@@ -1886,3 +1886,22 @@ def add_submitted_file(
         add_files_format_error(
             data, '"_files" is present in "submitted_answers" but is not an array'
         )
+
+
+ListItem = TypeVar("ListItem")
+
+
+def partition(
+    data: list[ListItem], pred: Callable[[ListItem], bool]
```

Comment:
```suggestion
    data: Iterable[ListItem], pred: Callable[[ListItem], bool]
```
Doesn't necessarily have to be a list. Type hint should probably be changed to `IterItem` or something.